src/ansys/templates/python/common/{{cookiecutter.__project_name_slug}}/.github/workflows/ci_cd.yml

```diff
@@ -66,57 +81,151 @@ jobs:
 
   doc-deploy-dev:
     name: "Deploy development documentation"
     runs-on: ubuntu-latest
-    needs: [build-library]
-    if: github.event_name == 'push' && !contains(github.ref, 'refs/tags')
+    needs: [doc-build, build-library]
+    if: github.ref == 'refs/heads/main'
     steps:
-      - uses: ansys/actions/doc-deploy-dev@v5
+      - uses: ansys/actions/doc-deploy-dev@v6
         with:
           cname: {{ '${{ env.DOCUMENTATION_CNAME }}' }}
           token: {{ '${{ secrets.GITHUB_TOKEN }}' }}
 
   doc-deploy-stable:
     name: "Deploy stable documentation"
     runs-on: ubuntu-latest
-    needs: [build-library]
+    needs: [doc-build, build-library]
     if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
     steps:
-      - uses: ansys/actions/doc-deploy-stable@v5
+      - uses: ansys/actions/doc-deploy-stable@v6
         with:
           cname: {{ '${{ env.DOCUMENTATION_CNAME }}' }}
           token: {{ '${{ secrets.GITHUB_TOKEN }}' }}
 
+  doc-index-dev:
+    name: "Deploy dev index docs"
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    needs: doc-deploy-dev
+    steps:
+      - name: "Deploy the latest documentation index"
+        uses: ansys/actions/doc-deploy-index@v6
+        with:
+          cname: "{{ '${{ env.DOCUMENTATION_CNAME }}' }}/version/dev"
+          index-name: "{{ '${{ env.LIBRARY_NAME }}'}}-vdev"
+          host-url: {{ '${{ env.MEILISEARCH_HOST_URL }}' }}
+          api-key: {{ '${{ env.MEILISEARCH_API_KEY }}' }}
+          python-version: {{ '${{ env.MAIN_PYTHON_VERSION }}' }}
 
+  doc-index-stable:
+    name: "Deploy stable docs index"
+    runs-on: ubuntu-latest
+    needs: doc-deploy-stable
+    steps:
+      - name: "Install Git and clone project"
+        uses: actions/checkout@v4
+
+      - name: "Install the package requirements"
+        run: pip install -e .
+
+      - name: "Get the version to PyMeilisearch"
+        run: |
+          VERSION=$(python -c "from {{ cookiecutter.__pkg_namespace }} import __version__; print('.'.join(__version__.split('.')[:2]))")
+          VERSION_MEILI=$(python -c "from {{ cookiecutter.__pkg_namespace }} import __version__; print('-'.join(__version__.split('.')[:2]))")
+          echo "Calculated VERSION: $VERSION"
+          echo "Calculated VERSION_MEILI: $VERSION_MEILI"
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          echo "VERSION_MEILI=$VERSION_MEILI" >> $GITHUB_ENV
+
+      - name: "Deploy the latest documentation index"
+        uses: ansys/actions/doc-deploy-index@v6
+        with:
+          cname: "{{ '${{ env.DOCUMENTATION_CNAME }}' }}/version/{{ '${{ env.VERSION }}' }}"
+          index-name: "{{ '${{ env.LIBRARY_NAME }}' }}-v{{ '${{ env.VERSION_MEILI }}' }}"
+          host-url: {{ '${{ env.MEILISEARCH_HOST_URL }}' }}
+          api-key: {{ '${{ env.MEILISEARCH_API_KEY }}' }}
+          python-version: {{ '${{ env.MAIN_PYTHON_VERSION }}' }}
```

Comment:
```suggestion
```
we add this after the lib become public, we can avoid indexing the private pages by meilisearch

Reply:
@Revathyvenugopal162 should I comment these out instead of deleting them?

Reply:
yes that works for me